src/crepe/system/InputSystem.cpp

```diff
@@ -0,0 +1,154 @@
+#include "../api/Button.h"
+#include "../api/EventManager.h"
+#include "ComponentManager.h"
+
+#include "system/InputSystem.h"
+
+using namespace crepe;
+
+void InputSystem::update() {
+	EventManager & event_mgr = EventManager::get_instance();
+	std::vector<SDLContext::EventData> event_list = SDLContext::get_instance().get_events();
+
+	for (const SDLContext::EventData & event : event_list) {
+		switch (event.event_type) {
+			case SDLContext::EventType::KEYDOWN: {
+				event_mgr.queue_event<KeyPressEvent>(KeyPressEvent{
+					.repeat = event.key_repeat,
+					.key = event.key,
+				});
+				break;
+			}
+			case SDLContext::EventType::KEYUP: {
+				event_mgr.queue_event<KeyReleaseEvent>(KeyReleaseEvent{
+					.key = event.key,
+				});
+				break;
+			}
+			case SDLContext::EventType::MOUSEDOWN: {
+				event_mgr.queue_event<MousePressEvent>(MousePressEvent{
+					.mouse_x = event.mouse_position.first,
+					.mouse_y = event.mouse_position.second,
+					.button = event.mouse_button,
+				});
+				last_mouse_down_position = event.mouse_position;
+				last_mouse_button = event.mouse_button;
+				break;
+			}
+			case SDLContext::EventType::MOUSEUP: {
+				MouseReleaseEvent mouse_release_event = MouseReleaseEvent{
+					.mouse_x = event.mouse_position.first,
+					.mouse_y = event.mouse_position.second,
+					.button = event.mouse_button,
+				};
+				event_mgr.queue_event<MouseReleaseEvent>(mouse_release_event);
+
+				// Calculate deltas for click detection
+				int delta_x = event.mouse_position.first - last_mouse_down_position.first;
+				int delta_y = event.mouse_position.second - last_mouse_down_position.second;
+
```

Comment:
Slechts een vraag uit nieuwsgierigheid:
Klopt het dat je zelf een MouseClickEvent hebt bedacht? En dat je een muisklik ziet als indrukken van de muisknop én weer loslaten binnen een gebied van `click_tolerance`? Dat is erg handig!

Reply:
ja die hebt ik zelf gemaakt want sdl heeft hier geen standaard functie voor